drivers/sensors/apds9922.c

```diff
@@ -0,0 +1,2598 @@
+   * are disabled until the data is read, no special action should be
+   * required to protect the work queue.
+   */
+
+  DEBUGASSERT(priv->work.worker == NULL);
+
+  ret = work_queue(HPWORK, &priv->work, apds9922_worker, priv, 0);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to queue work: %d\n", ret);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: apds9922_reset
+ *
+ * Description:
+ *   Reset the chip
+ *
+ * Input Parameters:
+ *   priv    - pointer to device structure
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_reset(FAR struct apds9922_dev_s *priv)
+{
+  int ret;
+
+  ret = apds9922_i2c_write8(priv, APDS9922_MAIN_CTRL, APDS9922_SW_RESET);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to reset the APDS9922\n");
+      return ret;
+    }
+
+  /* initialise setup to match the reset defaults etc. */
+
+  priv->als_setup.rate         = ALS_RATE100MS;
+  priv->als_setup.res          = ALS_RES200MS;
+  priv->als_setup.thresh.upper = ALS_DEF_THRESHU;
+  priv->als_setup.thresh.lower = ALS_DEF_THRESHL;
+  priv->als_setup.thresh_var   = ALS_DEF_VAR;
+  priv->als_setup.int_mode     = ALS_INT_MODE_THRESHOLD;
+  priv->als_setup.persistance  = ALS_DEF_PERSISTANCE;
+  priv->als_setup.als_factor   = 1;
+  priv->als_setup.range_lim    = 1;
+  priv->als_setup.autogain     = false;
+  priv->als_setup.channel      = ALS_VISIBLE;
+
+  priv->ps_setup.rate          = PS_RATE100MS;
+  priv->ps_setup.res           = PS_RES8;
+  priv->ps_setup.led_f         = PS_LED_FREQ60K;
+  priv->ps_setup.led_pk_on     = false;
+  priv->ps_setup.led_i         = PS_LED_CURRENT100MA;
+  priv->ps_setup.pulses        = PS_DEF_PULSES;
+  priv->ps_setup.thresh.upper  = PS_DEF_THRESHU;
+  priv->ps_setup.thresh.lower  = PS_DEF_THRESHL;
+  priv->ps_setup.cancel_lev    = PS_DEF_CANCEL_LVL;
+  priv->ps_setup.persistance   = PS_DEF_PERSISTANCE;
+  priv->ps_setup.notify        = PS_ALL_INFO;
+  priv->ps_setup.int_mode      = PS_INT_MODE_NORMAL;
+
+  /* Wait for device to power up properly after reset */
+
+  nxsig_usleep(50000);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: apds9922_probe
+ *
+ * Description:
+ *   Verify if sensor is present. Check if ID is 0xAB.
+ *
+ * Input Parameters:
+ *   priv    - pointer to device structure
+ *
+ * Returned Value:
+ *   Success or failure
+ *
+ ****************************************************************************/
+
+static int apds9922_probe(FAR struct apds9922_dev_s *priv)
+{
+  int ret;
+  uint8_t id = 0;
+
+  ret = apds9922_i2c_read8(priv, APDS9922_ID, &id);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to probe the APDS9922\n");
+      return ret;
+    }
+
+  if (id != APDS9922_ID_VAL)
+    {
+      snerr("ERROR: APDS9922 device ID is incorrect\n");
```

Comment:
```suggestion
      snerr("ERROR: APDS9922 device ID 0x%x is incorrect\n", id);
```
I think @acassis was talking about something like this